### DISCUSSION TITLE
Attribute Error in 'vtkCommonDataModelPython.vtkCellArray'

### DISCUSSION THREAD

**Discussion by chinei** [dipy/dipy#2824] [3. Community Lounge]
Hi All,

I'm facing a generic problem when wanting to use fury to visualize several different processing steps or results, such as to visuallize the DTI's diffusion tensor ellipsoids (see tutorial based example below), or the probability distribution function for each voxel etc.

More specifically, when running the following code example:

```python
from dipy.data import get_sphere
sphere = get_sphere('repulsion724')

from dipy.viz import window, actor

interactive = True

scene = window.Scene()

evals = dti_fit.evals[30:, 50:, ax_slice:ax_slice + 1]
evecs = dti_fit.evecs[30:, 50:, ax_slice:ax_slice + 1]

cfa = colored_FA[30:, 50:, ax_slice:ax_slice + 1]
cfa /= cfa.max()

scene.add(actor.tensor_slicer(evals, evecs, scalar_colors=cfa, sphere=sphere, scale=0.7))

print('Saving illustration as tensor_ellipsoids.png')
window.record(scene, n_frames=1, out_path='tensor_ellipsoids.png',
              size=(600, 600))
if interactive:
    window.show(scene)
```

I'm getting: AttributeError: `vtkCommonDataModelPython.vtkCellArray` object has no attribute `SetData`.

Has anyone faced the same problem and could provide me with a solution?

Thanks in advance

**Comment by skoudoro** [comment]
Can you provides your version of

- DIPY
- FURY
- VTK

thank you. I suppose you just need to update one of these libraries @chinei

**Reply by chinei** [reply]
Many thanks @skoudoro;

I have dipy version 1.5.0, fury version 0.8.0 and vtk version -PKG. When I use pip install vtk, it says: 

> Requirement already satisfied: vtk in c:\programdata\anaconda3...(-PKG-Version)

When I want to upgrade it using pip install vtk --upgrade, I get:

> ERROR: Cannot uninstall 'vtk'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.

The problem with distutils is described on the internet. However, they suggest to upgrade pip to version 10.0.0 [https://github.com/pypa/pip/issues/5247](url), but I already have version 20.0.2.

Any ideas?

**Reply by chinei** [reply]
For anybody interested in the solution:

Change vtk to version 9.1.0 (as the newest version (9.2.6) won't work either). For this, (and if you are using a windows environment) you need to force anaconda to remove vtk using: conda remove vtk adding the '--force' option. When using vtk version 9.2.6, trying to import vtk packages such as: `from dipy.viz import window, actor` you get: 

> ModuleNotFoundError: No module named 'vtkmodules.util'

So use version 9.1.0 and make sure to have the correct versions of dipy and fury, as outlined in the previous comment

**Comment by chinei** [comment]
issue solved
